File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        mp = defaultdict(int)
        for num in nums:
            mp[num] += 1
        mp = dict(sorted(mp.items(), key=lambda x: x[1], reverse=True))
        return list(mp.keys())[:k]

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] res = new int[nums.length];
        int product = 1;
        int count = 0;
        for (int num: nums) {
            if (num != 0) product *= num;
            else count++;
        }
        int i = 0;
        for (int num: nums) {
            if (count > 1) res[i++] = 0;
            else if (count == 1 && num == 0) res[i++] = product;
            else if (count == 1) res[i++] = 0;
            else res[i++] = product/num;
        }
        return res;
    }
}  
